src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java

```diff
@@ -0,0 +1,188 @@
+        len = Serialization.getNeedBytes(propertiesNames.size());
+        System.arraycopy(Serialization.encodeVi64(propertiesNames.size()), 0, bytes, idx, len);
+        idx += len;
+        for (String column : propertiesNames) {
+            len =  Serialization.getNeedBytes(column);
+            System.arraycopy(Serialization.encodeVString(column), 0, bytes, idx, len);
+            idx += len;
+        }
+
+        // 4. encode properties values
+        len = Serialization.getNeedBytes(propertiesValues.size());
+        System.arraycopy(Serialization.encodeVi64(propertiesValues.size()), 0, bytes, idx, len);
+        idx += len;
+        for (ObObj obj : propertiesValues) {
+            len =  obj.getEncodedSize();
+            System.arraycopy(obj.encode(), 0, bytes, idx, len);
+            idx += len;
+        }
+
+        return bytes;
+    }
+
+    /*
+     * Decode.
+     */
+    @Override
+    public Object decode(ByteBuf buf) {
+        // 0. decode header
+        super.decode(buf);
+
+        int len = (int) Serialization.decodeVi64(buf);
+        for (int i = 0; i < len; i++) {
+            String column = Serialization.decodeVString(buf);
+            rowKeyNames.add(column);
+        }
+
+        len = (int) Serialization.decodeVi64(buf);
+        for (int i = 0; i < len; i++) {
+            ObObj obj = new ObObj();
+            obj.decode(buf);
+            rowkey.add(obj);
+        }
+
+        len = (int) Serialization.decodeVi64(buf);
+        for (int i = 0; i < len; i++) {
+            String column = Serialization.decodeVString(buf);
+            propertiesNames.add(column);
+        }
+
+        len = (int) Serialization.decodeVi64(buf);
+        for (int i = 0; i < len; i++) {
+            ObObj obj = new ObObj();
+            obj.decode(buf);
+            propertiesValues.add(obj);
+        }
+
+        return this;
+    }
+
+    /*
+     * Get payload content size.
+     */
+    @Override
+    public long getPayloadContentSize() {
+        long payloadContentSize = 0;
+
+        payloadContentSize += Serialization.getNeedBytes(rowKeyNames.size());
+        for (String column : rowKeyNames) {
+            payloadContentSize += Serialization.getNeedBytes(column);
+        }
+
+        payloadContentSize += Serialization.getNeedBytes(rowkey.size());
+        for (ObObj obj : rowkey) {
+            payloadContentSize += obj.getEncodedSize();
+        }
+
+        payloadContentSize += Serialization.getNeedBytes(propertiesNames.size());
+        for (String column : propertiesNames) {
+            payloadContentSize += Serialization.getNeedBytes(column);
+        }
+
+        payloadContentSize += Serialization.getNeedBytes(propertiesValues.size());
+        for (ObObj obj : propertiesValues) {
+            payloadContentSize += obj.getEncodedSize();
+        }
+
+        return payloadContentSize;
+    }
+
+    public static ObTableSingleOpEntity getInstance(String[] rowKeyNames, Object[] rowKey,
```

Comment:
Maybe `generateEntity()` or `generateEntityFromKeys()` is a better name, since `getInstance()` sounds like we are trying to get a singleton.

Reply:
done